backend/common/utils/concurrency.py

```
@@ -27,28 +38,76 @@ def _env_int(name: str, default: int) -> int:
+# Recommended Railway values for 50 concurrent users:
+#   SIMULATION_MAX_STREAMS_PER_PROCESS=45
+#   SIMULATION_MAX_AI_CALLS_PER_PROCESS=25
+#
+# If experiencing hangs, reduce to 40/22
+# If stable but slow, can try increasing to 50/28 (monitor closely)
+# 
+# Performance note: Very low stream limits (e.g., 15) for many users (e.g., 50) cause
+# excessive queuing and perceived slowness. Better to allow more concurrency with
+# proper resource management than to over-restrict and queue everything.
+#
+# Recommended configuration for 2 replicas × 2.0 vCPU:
```

Ruff (0.14.8)

Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)